tests/integration/test_platforms.py

```diff
@@ -0,0 +1,60 @@
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Integration testing for platforms functionality.
+"""
+
+
+async def test_foo(flow, scheduler, run, mock_glbl_cfg, validate, monkeypatch):
+    global_conf = '''
+        [platforms]
+            [[myplatform]]
+                hosts = broken
+            [[anotherbad]]
+                hosts = broken2
+        [platform groups]
+            [[mygroup]]
+                platforms = myplatform, anotherbad'''
+    mock_glbl_cfg('cylc.flow.platforms.glbl_cfg', global_conf)
+
+    wid = flow({
+        "scheduling": {
+            "graph": {
+                "R1": "foo"
+            }
+        },
+        "runtime": {
+            "root": {},
+            "print-config": {
+                "script": "cylc config"
+            },
+            "foo": {
+                "script": "sleep 10",
+                "platform": "mygroup",
+                "submission retry delays": '3*PT5S'
+            }
+        }
+    })
+    validate(wid)
+    schd = scheduler(wid, paused_start=False, run_mode='live')
+    async with run(schd) as log:
```

Comment:
There is a traceback due to `FileNotFoundError` when running this test with `pytest -rA`. It seems benign but could potentially be distracting/confusing if trying to debug a test failure. No worries if it isn't easy to avoid.
```
Traceback (most recent call last):
  File "~/github/cylc-flow/cylc/flow/task_job_mgr.py", line 1187, in _prep_submit_task_job
    platform = get_platform(
               ^^^^^^^^^^^^^
  File "~/github/cylc-flow/cylc/flow/platforms.py", line 155, in get_platform
    return platform_from_name(task_conf['platform'], bad_hosts=bad_hosts)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/github/cylc-flow/cylc/flow/platforms.py", line 216, in platform_from_name
    platform_name = get_platform_from_group(
                    ^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/github/cylc-flow/cylc/flow/platforms.py", line 311, in get_platform_from_group
    raise NoPlatformsError(
cylc.flow.exceptions.NoPlatformsError: Unable to find a platform from group mygroup.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "~/github/cylc-flow/cylc/flow/task_events_mgr.py", line 118, in log_task_job_activity
    with open(os.path.expandvars(job_activity_log), "ab") as handle:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory:
    '~/cylc-run/cit-2024.../integration.test_platforms/test_foo/6d9d.../log/job/1/foo/NN/job-activity.log'
```